stadata/main.py

```diff
@@ -2,24 +2,140 @@
 import warnings
 import pandas as pd
 from tqdm import tqdm
+from functools import wraps
 import html
+import json
 from .material import Material
 
 BASE_URL = "https://webapi.bps.go.id/v1/"
 
+def validate_request(func):
+    """
+    Decorator to validate request
+
+    :param func: function to be decorated
+    :return: decorated function
+    """
+
+    @wraps(func)
+    def decorated(*args, **kwargs):
```

Comment:
I very love your propose to create universal function to interact with webapi. Unfortunately this function not working well

Reply:
May i know details on the result or error produced by the code?

Reply:
Below result when call client.view_publication('0000','0') :

`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/my-user/Documents/GitHub/stadata/stadata/main.py", line 567, in view_publication
    res = self.__get_view(domain=domain,model='publication',idx=id,lang='ind')
  File "/Users/my-user/Documents/GitHub/stadata/stadata/main.py", line 174, in __get_view
    return requests_model.view(self.TOKEN)
  File "/Users/my-user/Documents/GitHub/stadata/stadata/main.py", line 22, in decorated
    if args[0].TOKEN == '':
AttributeError: 'RequestModel' object has no attribute 'TOKEN'`